test/e2e/framework/scaletest/create-network-policies.go

```diff
@@ -0,0 +1,169 @@
+package scaletest
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	e2ekubernetes "github.com/microsoft/retina/test/e2e/framework/kubernetes"
+	yaml "gopkg.in/yaml.v2"
+
+	"github.com/microsoft/retina/test/e2e/framework/scaletest/templates"
+	netv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type CreateNetworkPolicies struct {
```

Comment:
Yes, that could be done with some refactoring.

kubernetes.CreateDenyAllNetworkPolicy step creates a specific NetworkPolicy (NP) that is encapsulated in a way where the only Namespace and one podSelector label could be provided.

scaletest.CreateNetworkPolicies step creates a number of NPs that are also specific with some podSelector labels needed for the scale-tests.

The first one would need to have more fields to allow more customization on generated NP. But it would also require to check and change the code everywhere it's currently being used. 

If it were done that way, how should this step that generates a single NP be reused to generate many NPs? Inside the ScaleTest function that generates the job (file test/e2e/jobs/scale.go) add a loop to add this same Step several times? Or a new step should be created encapsulating these multiple NP creations inside of it? (I'm not sure if the framework currently allows creation/reuse of steps inside of steps.)

Anyway, if this is the best approach, maybe this could be done in a separate refactoring task. Do you agree?

Reply:
I see the reasoning, while there might be a way to accommodate both scenarios, that may be out of scope for this pr given current limitations. Existing cases are verifying granular drop behavior unlike this suite, can shelve for later